### ISSUE TITLE
with no users logged in, sddm does not suspend session as configured in System Settings.

### ISSUE THREAD

**Issue by fduncanh** [sddm/sddm#1302]
When  kde5 System Settings is set to e.g. put system to sleep after x minutes of  inactivity,
this only works when the user is logged in.

with no users logged in, and the sddm login screen is on  the display, the screen correctly blanks 
the screen after inactivity, but never puts the system to sleep, if that is what is requested
in System Settings.

For consistency sddm should have also  have a feature   
allowing the system to be suspended (sleep/hibernate/hybrid sleep, etc)
if x minute has elapsed with no one logged in.

(obviously this would be off by default, for servers). And would be an "all-user" policy.

**Comment by lugdunum-1964** [comment]
I second this. Been looking for this feature for ages.

--
Lyon

**Comment by popanz** [comment]
Any news on this? It's supoptimal to see the login screen burns into your display.

**Comment by CV8R** [comment]
I third this - very frustrating.  As a relative newcomer to Kubuntu it took me ages to search and learn what was going on.

**Comment by YamashitaRen** [comment]
Fourth.
Here the screen is not even blanked.  Like @popanz said, having the login screen burn into the display is terrible. Not even talking about wasted energy...

**Comment by conromia** [comment]
This is a much needed "Feature" 
Now that SDDM is getting some more attention, this should be prioritised.
(As a workaround, I use GDM which I can freely edit time-outs in)

**Comment by Gebhartj** [comment]
Fifth.

**Comment by elydpg** [comment]
Sixth; I'm having the same issue on Kubuntu 23.10

**Comment by bakgwailo** [comment]
Seventh. Same issue on Arch Linux and Plasma 6rc2 and latest SDDM. Toddler tends to turn the computer on/off via physical buttons. When it restarts it just sits at the SDDM login in screen - big waste of energy that it doesn't go to sleep.

**Comment by erfan-star-1999** [comment]
Please add feature for set time to suspend in sddm i really need that

**Comment by Vogtinator** [comment]
Anyone wants to give https://github.com/sddm/sddm/pull/1878 a try?

This relies on logind (or consolekit) triggering a suspend on idle, so you might have to change `/etc/systemd/logind.conf` like

```
[Login]
IdleAction=suspend
IdleActionSec=5min
```

**Comment by erfan-star-1999** [comment]
> Anyone wants to give #1878 a try?
> 
> This relies on logind (or consolekit) triggering a suspend on idle, so you might have to change `/etc/systemd/logind.conf` like
> 
> ```
> [Login]
> IdleAction=suspend
> IdleActionSec=5min
> ```

Not Worked

**Comment by kelvie** [comment]
I implemented a workaround using swayidle and whatever wayland compositors swayidle supports, adapt and test to your leisure 

create a file in `/var/lib/sddm/compositor` (make sure the permissions are locked down as I think this runs as root).

```bash
#!/bin/bash

kwin_wayland --drm --no-lockscreen --no-global-shortcuts --locale1 &
kwin_pid=$!

# sleep is fine since this just adds 5 seconds to the idle timeout,
# and we need to wait for kwin to start up
sleep 5
swayidle -w timeout 300 'systemctl suspend' &
swayidle_pid=$!

trap kill_children SIGTERM
function kill_children {
    kill -TERM $kwin_pid
    kill -TERM $swayidle_pid
    wait
    exit 0
}

wait
```

Run:
```
sudo chmod +x /var/lib/sddm/compositor
```

And in `/etc/sddm.conf.d/10-wayland.conf`:

```conf
[General]
DisplayServer=wayland

[Wayland]
CompositorCommand=./compositor
```

The `kwin` commandline is from the Arch Wiki page for SDDM.

With this, the screen will go to sleep in 5 minutes of idle.

I'm not entirely sure the SIGTERM trap is needed, but I added it just in case.

**Comment by gillesgagniard** [comment]
> Anyone wants to give [#1878](https://github.com/sddm/sddm/pull/1878) a try?
> 
> This relies on logind (or consolekit) triggering a suspend on idle, so you might have to change `/etc/systemd/logind.conf` like
> 
> ```
> [Login]
> IdleAction=suspend
> IdleActionSec=5min
> ```

Thanks a lot @Vogtinator ! Your patch works perfectly.

For Arch users, I've published a sddm alternative package which applies your PR: https://aur.archlinux.org/packages/sddm-idle

**Comment by EtienneK** [comment]
This would be a really helpful feature. My poor screen is burning.

**Comment by KayTaneJupiter** [comment]
Has this been fixed yet? More than stunned this bug has been outstanding for almost 5 years ...

**Comment by kbkozlev** [comment]
It is still an issue in Fedora 42 KDE . Any news?

**Comment by p-gonzo** [comment]
This is still an issue on Fedora 43 with KDE Plasma 6.5.4.

**Comment by Robula** [comment]
I support this feature, my new OLED displays SDDM for hours on end beautifully, but I'd rather the system turn off the display or just go to sleep.

**Comment by sdirek** [comment]
Hi everyone,

Like many of you, I've been incredibly frustrated by SDDM keeping the monitors on indefinitely before login. Since there is still no native power management out of the box, I decided to build a permanent workaround that actually solves the root problems without breaking the system.

Instead of relying on fragile display-server hacks that often trigger Wayland crash loops or NVIDIA black screens on resume, I wrote a lightweight systemd service (`kernel-idle`).

Here is how it solves the issue:
* It safely turns off the display and suspends the system directly at the hardware/kernel level while sitting on the SDDM screen.
* It automatically reads your existing KDE Plasma power settings (so you don't even need to configure it).
* It survives Wayland/NVIDIA quirks by handling VT switching and backlight power properly before sleeping.
* The second you log in, it gracefully shuts up and hands power management back to KDE Plasma.

To avoid spamming this thread with a wall of code, I've put the script, the systemd service file, and quick installation instructions in a dedicated repo here: 

👉 https://github.com/sdirek/kernel-idle

I hope this finally gives some peace of mind to anyone still dealing with glowing monitors at night! Let me know if it works for your setup.

**Comment by Vogtinator** [comment]
The code above from @sdirek looks AI generated, so be cautious.

There is already https://github.com/sddm/sddm/pull/1878, which should be sufficient.

**Comment by sdirek** [comment]
> The code above from [@sdirek](https://github.com/sdirek) looks AI generated, so be cautious.
> 
> There is already [#1878](https://github.com/sddm/sddm/pull/1878), which should be sufficient.

Yeah I did use AI to help write the bash code itself. But the logic, edge cases and all the hardware testing was done manually by me.

I actually tried #1878 first before doing any of this, but it just didn't work on my setup at all. That's the whole reason I built this. Since the existing solutions didn't fix the issue for me, I had to put together a workaround that actually turns off my screen right now.

Just sharing the repo here in case #1878 isn't working for someone else either.
